src/Core/Auth/Models/Api/Request/Accounts/RegisterFinishRequestModel.cs

```diff
@@ -50,15 +63,28 @@ public class RegisterFinishRequestModel : IValidatableObject
 
     public User ToUser()
     {
+        // PM-28143 - Remove line below
+        // When we process this request to a user object, check if the unlock and authentication
+        // data has been passed through, and if so they should have matching values.
+        MasterPasswordUnlockData.ThrowIfExistsAndNotMatchingAuthenticationData(MasterPasswordAuthenticationData, MasterPasswordUnlockData);
+
+        // PM-28143 - Remove line below
+        MasterPasswordAuthenticationData.ThrowIfExistsAndHashIsNotEqual(MasterPasswordAuthenticationData, MasterPasswordHash);
+
         var user = new User
         {
             Email = Email,
             MasterPasswordHint = MasterPasswordHint,
-            Kdf = Kdf,
-            KdfIterations = KdfIterations,
-            KdfMemory = KdfMemory,
-            KdfParallelism = KdfParallelism,
-            Key = UserSymmetricKey,
+            Kdf = MasterPasswordUnlockData?.Kdf.KdfType ?? Kdf
+                ?? throw new Exception("KdfType couldn't be found on either the MasterPasswordUnlockData or the Kdf property passed in."),
```

Comment:
❓  We already handle the validation in `Validate` and this is request model, so the expectation should be clear, that `Validate` is always called first. Is there any point in doing it twice ?
```suggestion
            Kdf = MasterPasswordUnlockData?.Kdf.KdfType ?? Kdf!
```